src/components/dialog/bl-dialog.ts

```diff
@@ -55,7 +55,7 @@ export default class BlDialog extends LitElement {
 
   updated(changedProperties: PropertyValues<this>) {
     if (changedProperties.has('open')) {
```

Comment:
I think we can just check if it’s really changed here to take an action:
```suggestion
    if (changedProperties.has('open') && changedProperties.get('open') !== this.open) {
```

In this case we would not need to further checks in `toggleHandler`.

Reply:
I tried this solution before but unfortunately, it didn't solve the close-event firing issue

Reply:
Have you tried the following condition? 

` if (changedProperties.has('open') && changedProperties.get('open') !== undefined) `

I think the reason why @muratcorlu's suggestion doesn't work is that the property is assigned as undefined in phase of component getting attached to the document's DOM.

Reply:
You are right @olkeoguz but we missed one thing according to your solution first open event won't trigger

You can try this:

```suggestion
    if (changedProperties.has('open') && (changedProperties.get('open') !== undefined || this.open)) {
```

Reply:
Yes that fixed the issue, thanks all

Reply:
Please make sure this prevents running toggle handler multiple time in case of setting same value again, like:

```js
const dialog = document.querySelector(‘bl-dialog’);

dialog.open = true;

// Below should not do any change
dialog.open = true;
```

or

```js
const dialog = document.querySelector(‘bl-dialog’);

dialog.open = false;

// Below should not do any change
dialog.open = false;
```